main.cpp

```diff
@@ -214,291 +256,356 @@ struct MacbookPro
+{
+    if (hardDriveSpaceAvailable > app.sizeInGigabytes)
+    {
+        std::cout << "Downloading app: " << app.name << std::endl;
+    }
+    std::cout << "Not enough space to download app" << std::endl;
+}
+
+
  struct Iphone
  {
-    // 1) size of screen
     float diagScreenSizeInches = 5.85f;
-    // 2) model number
     std::string model = "iPhone 11 Pro";
-    // 3) number of call minutes used this month
     float monthlyCallMinsUsed = 108.3f;
-    // 4) time spent browsing instagram
     double minSpentBrowsingInstagramToday = 34.5;
-    // 5) number of side buttons
     int numSideButtons = 3;
 
-    // 1) make calls
     float makeCall(std::string nameOfCallee);
-    // 2) send texts
     void sendText(std::string message);
-    // 3) set a timer
     void setTimer(double duration);
-
  };
 
-/*
-Thing 4) corgi
-5 properties:
-    1) number of legs
-    2) weight
-    3) average loudness of bark in dB
-    4) color of eyes
-    5) age
-3 things it can do:
-    1) bark
-    2) beg for treats
-    3) play catch
- */
+float Iphone::makeCall(std::string nameOfCallee)
+{
+    if (nameOfCallee != "Bob Smith")
+    {
+        std::cout << "Calling " << nameOfCallee << " for a short chat." << std::endl;
+        return 20.f; // should probably make a new variable to track how long the call is
+    }
+    std::cout << "I don't feel like talking to Bob right now." << std::endl;
+    return 0.f;
+}
+
+void Iphone::sendText(std::string message)
+{
+    std::cout << "Sending message: " << message << std::endl;
+}
+
+void Iphone::setTimer(double duration)
+{
+    std::cout << "Set timer for " << duration << " mins." << std::endl;
+}
+
+
 struct Corgi
 {
-    // 1) number of legs
     int numLegs = 4;
-    // 2) weight
     float weightInPounds = 27.2f;
-    // 3) average loudness of bark in dB
     float avgBarkLoudnessDb = 82.1f;
-    // 4) color of eyes
     std::string eyeColor = "Dark brown";
-    // 5) age   
     int age = 7;
 
-    // 1) bark
     void bark(float loudness);
-    // 2) beg for treats
     void begForTreats(int numOfTreats, float durationOfBeggingInMin = 248.2f);
-    // 3) play catch
-    void playCatch(int numTimesThrowBall);
+    void playCatch(unsigned int numTimesThrowBall);
 };
 
- /*
-Thing 5) VR: External camera
-5 properties:
-    1) amount of megapixels
-    2) diameter in mm
-    3) resolution
-    4) color mode
-    5) aperture
-3 things it can do:
-    1) scan room to create play boundary
-    2) detect dangerous obstacles in physical space
-    3) 'pass through' view while wearing headset
- */
- struct VRCamera
- {
-    // 1) amount of megapixels
+void Corgi::bark(float loudness)
+{
+    if (loudness > 10.f)
+    {
+        std::cout << "Corgi barked loud" << std::endl;
+    }
+    else if (loudness > 50.f)
+    {
+        std::cout << "Corgi barked REALLY loud!" << std::endl;
+    }
+    std::cout << "I barely heard anything" << std::endl;
```

Comment:
no matter what the `loudness` is set to, `"I barely heard anything"` will still print.
you need an `else` statement here:
```
if( loudness > 10.f ) { ... }
else if( loudness > 50.f { ... }
else { I barely heard anything }
```
Furthermore, `loudness > 10` will catch when the `loudness is > 50`.  You need to check that condition before checking if loudness is > 10:
```
if loudness > 50 ...
else if loudness > 10 ...
else ...
```
- [ ] revise